src/coreclr/jit/importercalls.cpp

```diff
@@ -3635,6 +3635,30 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                     }
                 }
 
+                // Load handle directly for known types
+                if (retNode == nullptr)
+                {
+                    bool                 isExact = false;
+                    bool                 notNull = false;
+                    CORINFO_CLASS_HANDLE typeHnd = gtGetClassHandle(op1, &isExact, &notNull);
+                    if ((typeHnd != NO_CLASS_HANDLE) && isExact &&
+                        ((info.compCompHnd->getClassAttribs(typeHnd) & (CORINFO_FLG_SHAREDINST | CORINFO_FLG_GENERIC_TYPE_VARIABLE)) != 0))
```

Comment:
```suggestion
                        ((info.compCompHnd->getClassAttribs(typeHnd) & CORINFO_FLG_SHAREDINST) != 0))
```
You should never see generic variables here. It does not hurt to check for `CORINFO_FLG_GENERIC_TYPE_VARIABLE` here, but it is unnecessary.

Reply:
I'm currently trying to check if this will make the CI pass cause I'm getting some weird failures that seem to look like `gtGetClassHandle` is returning Canon for shared generics, so this is a temporary check to see if the CI will pass.

Reply:
> `gtGetClassHandle` is returning Canon for shared generics

Is that a problem? I would expect `gtGetClassHandle` to return shared generic types in some cases, with exact=true.

Reply:
> I would expect `gtGetClassHandle` to return shared generic types in some cases, with exact=true.

I'd rather expect isExact to be false in such case, nevertheless seems like the failures on CoreCLR disappeared after I've added the `CORINFO_FLG_GENERIC_TYPE_VARIABLE` check so it looks like `gtGetClassHandle` seems to return generic variables as exact types? (there's some different failure on NativeAOT that's still there)

Reply:
> I'd rather expect isExact to be false in such case

It would be a de-optimization. For example, consider `var d = new Dictionary<Canon, Canon>`. I expect `isExact` is `true` in this case. The type of `d` is exact in the sense that we can still perform optimizations like de-virtualization.

Reply:
> failures on CoreCLR disappeared after I've added the CORINFO_FLG_GENERIC_TYPE_VARIABLE

That sounds odd. It would be useful to find out where the generic variable is coming from in this case.